clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1436,7 +1443,7 @@ enable_tracing_instrumentation()
             event_bb_instru2instru, &memtrace_pri))
         DR_ASSERT(false);
     dr_register_filter_syscall_event(event_filter_syscall);
-    tracing_enabled = true;
+    dr_atomic_add32_return_sum(&tracing_enabled, 1);
```

Comment:
Is there any benefit to using `dr_atomic_store32` instead of `dr_atomic_add32_return_sum`? Former may be less expensive I guess.

Reply:
As `tracing_enabled` is just a boolean that makes a one-way transition to `true`, may be better to use former.

Reply:
Switched to store.